user/super/com/google/gwt/emul/java/util/stream/DoubleStream.java

```diff
@@ -239,5 +277,19 @@ static DoubleStream of(double t) {
 
   DoubleSummaryStatistics summaryStatistics();
 
+  default DoubleStream takeWhile(DoublePredicate predicate) {
```

Comment:
```
DoubleStream.iterate(0.0, d -> d + 2.0)
        .takeWhile(d -> d < 50)
        .boxed()
        .collect(Collectors.toList());
```
should terminate, but with `filter` it won't

Reply:
Ah thank you, good catch. Ill add the example code you provided to ensure that we stop with infinite streams.

Reply:
Been unable to work on this for some time, and this next iteration of the work took longer than I had intended, but I think the added tests confirm that I'm on the right track now, PTAL.

I'm going to look into your other note the next chance I get.